lib/Dialect/D2M/Transforms/GenericLowerDMAs.cpp

```diff
@@ -340,7 +340,28 @@ class D2MGenericLowerAffineDMAsRewritePattern : public OpRewritePattern<DMAOp> {
     size_t coalescingFactor = calculateCoalescingFactor(
         memoryMap, memrefGridShape, memrefShardShape, elemSizeBytes);
 
-    return {streamIndices, coalescingFactor};
+    //--------------- trial analytical coalescing approach --------------//
+    auto altCoalescingFactor = ttmlir::utils::analyzeShardDimContiguity(
+        memoryMap, memref.getShape(), memrefGridShape.size(),
+        memoryMap.getNumResults() - 1, elemSizeBytes);
+    llvm::dbgs() << "[CoalescingFactor] alt = " << altCoalescingFactor << " vs "
+                 << coalescingFactor << "\n";
+
+    if (coalescingFactor % size_t(altCoalescingFactor) != 0) {
```

Comment:
:warning: **google\-readability\-casting** :warning:
C\-style casts are discouraged; use static\_cast
```suggestion
    if (coalescingFactor % static_cast<size_t>(altCoalescingFactor) != 0) {
```